src/ranch_listener_sup.erl

```diff
@@ -32,7 +32,12 @@ start_link(Ref, Transport, TransOpts, Protocol, ProtoOpts) ->
 -spec init({ranch:ref(), module(), module(), module()})
 	-> {ok, {supervisor:sup_flags(), [supervisor:child_spec()]}}.
 init({Ref, Transport, Protocol, Logger}) ->
-	ok = ranch_server:set_listener_sup(Ref, self()),
+	case put(running, true) of
```

Comment:
Less happy about this one. Can't we move the call to `start_link` (and call only when it succeeds, as a result)? Or it's really important to have it there?

Reply:
We can't move it to `start_link` because it has to be done in case of a restart, otherwise the old pid will remain in `ranch_server` and the new one will never make it there.

Reply:
Nonsense, forget what I said =^^=

If we move it to start_link, the difference is that the listener is registered in ranch_server _after_ it has been started, instead of when it is about to be started.

The only scenario I can think of where this matters is, when an acceptor has been started, accepts a connection and starts a protocol, and that protocol would call ranch_server:get_listener_sup, before the full initialization is done and the listener has been registered. This is a very obscure corner case, and I don't think we need to have any concerns about it, right?

Reply:
`ranch_server` itself is not documented so this isn't an issue. The only relevant function is `ranch:remove_connection/1` and I suppose it could happen that the listener is not yet fully started when this function is called. So I guess this could indeed be an issue.

Reply:
Ok, so we need to do this here after all...

Reply:
We need a test that connects and the protocol code removes the connection immediately to try to see if the aforementioned race condition occurs.

Reply:
Something like `acceptors_SUITE:tcp_remove_connections`?

Reply:
Except without waiting for the listener to be up, or during an upgrade.

Reply:
Glad to see that this one was not necessary after all!!